app/services/CompositeAdminUnitRegisterService.scala

```diff
@@ -0,0 +1,19 @@
+package services
+import uk.gov.ons.sbr.models.UnitKey
+import uk.gov.ons.sbr.models.unitlinks.UnitType.{ PayAsYouEarn, ValueAddedTax }
+
+import scala.concurrent.Future
+
+class CompositeAdminUnitRegisterService(vatRegisterService: UnitRegisterService, payeRegisterService: UnitRegisterService) extends UnitRegisterService {
+  override def isRegisteredUnit(unitKey: UnitKey): Future[UnitRegisterResult] = {
+    require(unitKey.unitType == ValueAddedTax || unitKey.unitType == PayAsYouEarn, s"Unsupported unit type [${unitKey.unitType}]")
+    unitKey.unitType match {
+      case ValueAddedTax => vatRegisterService.isRegisteredUnit(unitKey)
+      case PayAsYouEarn => payeRegisterService.isRegisteredUnit(unitKey)
+      /*
+       * catch-all to prevent compiler warning that match is not exhaustive - but this should not be hit
```

Comment:
Might it be worth having a `trait` for editable unit types to solve this issue?

Reply:
That is an interesting idea - I suppose cases like this do indicate that the type system is not being used to its full extent.

The problem here would be that ```UnitRegisterService``` is a general purpose trait (of which there are other implementations).  The function takes a ```UnitKey``` which has a ```UnitType``` not an "editable unit type" - so as it currently stands you would have to match on the more general unit type.  I suppose you could have a ```UnitType``` -> "editable unit type" partial function which is called prior to matching but this itself feels a bit ugly.

 In general, I'm currently trying to let edit evolve in a simple and relatively unconstrained way, as we know that we are currently only scratching the surface of edit requirements, and it is inevitable that we will need to add more structure as the more complicated requirements are implemented.